apps/wallet/src/ui/app/shared/social-buttons/BaseSocialButton.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { cva, type VariantProps } from 'class-variance-authority';
+import { forwardRef, type Ref, type FunctionComponent, type SVGProps } from 'react';
+import { ButtonOrLink, type ButtonOrLinkProps } from '../utils/ButtonOrLink';
+import { Text } from '_app/shared/text';
+
+const styles = cva(
+	'w-full cursor-pointer rounded-xl inline-flex items-center justify-center gap-3 px-4 py-2 disabled:opacity-40 focus:opacity-80',
+	{
+		variants: {
+			variant: {
+				outline:
+					'bg-white text-steel-dark border border-solid border-steel hover:border-steel-dark',
+				facebook: 'bg-[#1877F2] border-none text-white',
```

Comment:
Feels weird to me to have these brand variants but then not just manage the variable icons and all of the types of buttons here directly. It's like a hybrid of a generic component and one that knows about brand variants, I'd rather it just lean in and do all the branding and not need the dedicated components for each brand that you create.

Reply:
yep I totally agree, I actually did exactly that in one iteration of this but ended up changing it 😆

I'll update this!